app/store/services/login.js

```diff
@@ -1,24 +1,33 @@
 // constants
-import { API_ENDPOINT, APPLICATION_TYPE, RESOURCES, METHODS } from '../constants/rest';
+import { API_ENDPOINT, RESOURCES, METHODS } from '../constants/rest';
 
 // helper
 import { callJSONApi } from './helpers/api';
 
 
-export const fetchToken = (formData) => {
-  const headers = new Headers();
-  headers.append('Accept', 'application/json');
-  headers.append('Origin', '*');
-  headers.append('Cache-Control', 'no-cache');
-  headers.append('Content-Type', APPLICATION_TYPE.FORM_DATA);
+export const authenticate = (temporaryAccessToken) => {
+  // TODO: use config file to retrieve client id + client secret.
+  const formData = new FormData();
+  formData.append('client_id', '15688');
+  formData.append('client_secret', '');
+  formData.append('code', temporaryAccessToken);
 
   const request = {
     endpoint: {
       url: API_ENDPOINT + RESOURCES.OAUTH,
       httpVerb: METHODS.POST,
     },
-    headers,
-    body: formData,
+    parameters: formData,
   };
-  return callJSONApi(request);
+
+  return callJSONApi(request)
+    .then(
+      response => ({
+        token: response.data.access_token,
+        athleteID: response.data.athlete.id,
+        athleteDetails: {
+          firstname: response.data.athlete.firstname,
+          lastname: response.data.athlete.lastname } }),
+      error => ({ error }),
+    );
 };
```

Comment:
i think it could be better here. May be we could extract mapping/selector of the input and the output?
not sure :thinking: